tests/test_file_utils.py

```diff
@@ -338,3 +363,198 @@ def test_get_endian_resets_the_file_pointer(self):
+        assert sandbox.problems
+
+    def test_mkdir(self, sandbox: FileSystem):
+        sandbox.mkdir(Path("directory"))
+
+        assert (sandbox.root / "directory").is_dir()
+        assert sandbox.problems == []
+
+    def test_mkdir_outside_sandbox(self, sandbox: FileSystem):
+        sandbox.mkdir(Path("../directory"))
+
+        assert not (sandbox.root / "../directory").exists()
+        assert sandbox.problems
+
+    def test_mkfifo(self, sandbox: FileSystem):
+        sandbox.mkfifo(Path("named_pipe"))
+
+        assert (sandbox.root / "named_pipe").is_fifo()
+        assert sandbox.problems == []
+
+    def test_mkfifo_outside_sandbox(self, sandbox: FileSystem):
+        sandbox.mkfifo(Path("../named_pipe"))
+
+        assert not (sandbox.root / "../named_pipe").exists()
+        assert sandbox.problems
+
+    def test_create_symlink(self, sandbox: FileSystem):
+        sandbox.create_symlink(Path("target file"), Path("symlink"))
+
+        output_path = sandbox.root / "symlink"
+        assert not output_path.exists()
+        assert os.readlink(output_path) == "target file"
+        assert sandbox.problems == []
+
+    def test_create_symlink_absolute_paths(self, sandbox: FileSystem):
+        sandbox.write_bytes(Path("target file"), b"test content")
+        sandbox.create_symlink(Path("/target file"), Path("/symlink"))
+
+        output_path = sandbox.root / "symlink"
+        assert output_path.exists()
+        assert os.readlink(output_path) == "target file"
+        assert sandbox.problems == []
+
+    def test_create_symlink_absolute_paths_self_referenced(self, sandbox: FileSystem):
+        sandbox.mkdir(Path("/etc"))
+        sandbox.create_symlink(Path("/etc/passwd"), Path("/etc/passwd"))
+
+        output_path = sandbox.root / "etc/passwd"
+        assert not output_path.exists()
+        assert os.readlink(output_path) == "../etc/passwd"
+        assert sandbox.problems == []
+
+    def test_create_symlink_outside_sandbox(self, sandbox: FileSystem):
+        sandbox.create_symlink(Path("target file"), Path("../symlink"))
+
+        output_path = sandbox.root / "../symlink"
+        assert not os.path.lexists(output_path)
+        assert sandbox.problems
+
+    def test_create_symlink_path_traversal(
+        self, sandbox: FileSystem, sandbox_parent: Path
+    ):
+        """Document a remaining path traversal scenario through a symlink chain.
+
+        unblob.extractor.fix_symlinks() exists to cover up cases like this.
+        """
+        (sandbox_parent / "outer-secret").write_text("private key")
+
+        # The path traversal is possible because at the creation of "secret" "future" does not exist
+        # so it is not yet possible to determine if it will be a symlink to be allowed or not.
+        # When the order of the below 2 lines are changed, the path traversal is recognized and prevented.
+        sandbox.create_symlink(Path("future/../outer-secret"), Path("secret"))
```

Comment:
I initially felt that this should have been handled by the Sandbox, but since we still need to call `fix_extracted_directory` and `fix_symlink` for third party extractors, I'm okay with the way things are. Just noting it for further references.